src/utils/components/DiskModal/utils/helpers.ts

```diff
@@ -226,3 +228,23 @@ export const getRemoveHotplugPromise = (vm: V1VirtualMachine, diskName: string)
   };
   return removeVolume(vm, bodyRequestRemoveVolume);
 };
+
+export const getDisksFromVMAndVMI = (
+  vm: V1VirtualMachine,
+  vmi: V1VirtualMachineInstance,
+): V1Disk[] => {
+  const vmiDisksNames = (vmi?.spec?.domain?.devices?.disks || [])?.map((disk) => disk?.name);
+  const missingDisksFromVMI = getDisks(vm)?.filter((disk) => !vmiDisksNames?.includes(disk?.name));
+  return [...(vmi?.spec?.domain?.devices?.disks || []), ...(missingDisksFromVMI || [])];
+};
+
+export const getVolumesFromVMAndVMI = (
```

Comment:
`get<something>FromVMAndVMI` is a slippery slope, looks good here but next PR it will be `get<X>FromVMLikeObjcts` :-)
Can we do a `getMissingDisksFromVMI` and add them at  the bottom of the disks list ?
it will be the same thing, that is what you do in line 238, add the missing to the end of the list, but we will avoid the need for a `fromVMandVMI` method

Reply:
changed, can you take another look?